cluster/images/provider-terraform-controller/Dockerfile

```diff
@@ -1,16 +0,0 @@
-FROM BASEIMAGE
-
-ARG ARCH
-ARG TINI_VERSION
-
-ADD provider /usr/local/bin/crossplane-terraform-provider
-ADD .gitconfig .gitconfig
-
-# As of Crossplane v1.3.0 provider controllers run as UID 2000.
-# https://github.com/crossplane/crossplane/blob/v1.3.0/internal/controller/pkg/revision/deployment.go#L32
-RUN mkdir /tf
-RUN chown 2000 /tf
```

Comment:
ah good catch! Was a bit too eager deleting earlier, but it looks relevant based on the history so I'm going to bring it back.

cc @negz

Reply:
@jastang SGTM -- feel free to ping we when updated :)

Reply:
@hasheddan done!

Reply:
> I'm not actually sure what the purpose of this /tf directory is,

Just for completeness in case anyone finds this - the /tf directory is used by the provider as a workspace for terraform - a directory for each Workspace object is created under /tf using the UUID of the Workspace object, and that's where terraform init/apply/etc are executed for that Workspace object.  We could move it to /var/tmp/tf or somewhere else "inside" the image if we want to make the Dockerfile more generic.